core/app/models/spree/address.rb

```diff
@@ -84,6 +84,7 @@ def self.immutable_merge(existing_address, new_attributes)
     # @return [Hash] hash of attributes contributing to value equality with optional merge
     def self.value_attributes(base_attributes, merge_attributes = {})
       base = base_attributes.stringify_keys.merge(merge_attributes.stringify_keys)
+      base.delete("name") unless Spree::Config.use_combined_first_and_last_name_in_address
```

Comment:
if `use_combined_first_and_last_name_in_address` is false, if I recall correctly `name` should not be part of the attributes provided in an address, but maybe I'm missing something.

Reply:
We have custom forms that only have `firstname` and `lastname` as attributes. But the same is true in `solidus_frontend` v2.11.14 https://github.com/solidusio/solidus/blob/v2.11/frontend/app/views/spree/address/_form.html.erb#L8-L18 btw. 

Somehow Solidus is setting `name` before we reach this code and if `name` is given it will not be updated. See L97-99 below. Maybe there is a better fix, but this is the easiest with the least impact that fixes this issue. Credit to @cpfergus1